### PR TITLE
Move all children in <Transition />

### LINKED ISSUE #682: React Pose: PoseGroup does not animate elements before the one being removed
## Describe the bug
Not all elements are being animated when a PoseGroup child is removed / added

## How to reproduce
1. Go to https://codesandbox.io/s/m4j1py31xx
2. Click on one of the middle elements
3. You can see elements on the right smoothly translating to their new position while elements on the left jump to their new position without animation

This behavior is even more obvious when clicking on the leftmost or rightmost element

## Expected behaviour
I'd expect all elements to be transitionned to their new position

## Device information
In which environment are you reproducing the bug?
 - OS: macOS
 - Browser: chrome, safari
 - Version 4.0.4

### PR DESCRIPTION
**Pull request by esquevin**
Fixes #682

### CODE REVIEW

File: packages/react-pose/src/components/Transition/children.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ const handleTransition = (
       }
 
       const nextIndex = nextKeys.indexOf(key);
-      return nextIndex !== -1 && i !== nextIndex;
+      return nextIndex !== -1;
```
Comment (InventingWithMonster) [inline packages/react-pose/src/components/Transition/children.ts]: Is the resulting code any different to `leaving.indexOf(key) !== -1` basically saying, if it isnt entering or leaving?

**Comment by Andarist** [comment]
Would you be able to write a test for this one? Not sure how hard it is at the moment though.

**Comment by esquevin** [comment]
I've added a test and slightly changed the code to better reflect that `moving = not (entering or leaving)`

Please take another look.

**Comment by InventingWithMonster** [comment]
We might need to optimise this approach for longer lists, but IMO looks good to me

**Pull request merged**